google-cloud-pubsub/lib/google/cloud/pubsub/publish_result.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Pubsub
+      ##
+      # # PublishResult
+      #
+      class PublishResult
```

Comment:
We feel the encapsulation has value and is what we have done everywhere. It protects users from changes to the Low Level API, which has experienced more changes than you may expect.